File: solutions/2971. Find Polygon With the Largest Perimeter/2971.py
# Time complexity: O(sort)
# Space complexity: O(sort)
class Solution:
  def largestPerimeter(self, nums: List[int]) -> int:
    nums.sort()
    prefix = 0
    ans = -1

    for num in nums:
      if num < prefix:
        ans = num + prefix

      prefix += num

    return ans   

File: solutions/2971. Find Polygon With the Largest Perimeter/2971.cpp
// Time complexity: O(sort)
// Space complexity: O(sort)
class Solution {
public:
  long long largestPerimeter(vector<int>& nums) {
    ranges::sort(nums);
    long long previousElementsSum = 0;
    long long ans = -1;
    for (int num : nums) {
      if (num < previousElementsSum)
        ans = num + previousElementsSum;

      previousElementsSum += num;
    }

    return ans;
  }
};